### PR TITLE
Make progress bar output tests independent of formatting details

### PR DESCRIPTION
**Pull request by jdemaeyer**
Our tests for the progress bar output during custom image building/pushing sometimes randomly fail. This is because the output of carriage returns, newlines, and ANSI escape sequences is dependent on the machine that runs the test.

For example, [this test](https://ci.appveyor.com/project/scrapinghub/shub/build/master-554/job/5e7mabanrd3cppe4#L138) failed because the actual output was `\x1b[A\\n\\x1b[AThe image registry/user/project:test pushed successfully.` while the expected output was `\\n\\nThe image registry/user/project:test pushed successfully.`

This PR should make the tests consistent by stripping these characters before checking the expected output.

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -69,3 +70,8 @@ def format_expected_progress(progress):
         for sym in to_replace:
             progress = progress.replace(sym, to_replace[sym])
     return progress
+
+
+def clean_progress_output(output):
+    """Return output cleaned from \r, \n, and ANSI escape sequences"""
+    return re.sub('\n|\r|(\x1b\[|\x9b)[^@-_]*[@-_]|\x1b[@-_]', '', output)
```
Comment (chekunkov) [inline tests/utils.py]: the regex is hard to follow. can you visually split it and add comments using https://docs.python.org/2/library/re.html#re.X ?

Comment (jdemaeyer) [inline tests/utils.py]: That was actually a lot of fun, thanks :D

Comment (jdemaeyer) [inline tests/utils.py]: uhh nevermind, I seem to have broken it

Comment (jdemaeyer) [inline tests/utils.py]: fixed
